server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -761,7 +774,6 @@ private boolean verifyManifestAndClusterState(ClusterMetadataManifest manifest,
         private boolean shouldWriteFullClusterState(ClusterState clusterState) {
             if (lastAcceptedState == null
                 || lastAcceptedManifest == null
-                || lastAcceptedState.term() != clusterState.term()
```

Comment:
there are more edge cases that need to handled here
1. RemotePublicationEnabled = true for active CM1, all nodes have latest local and remote state
2. RemotePublicationEnabled = false for active CM1, now lastManifest and lastState will be stale in RemotePersistedState in CM2 (as publishRequest.manifest would be null)
3. CM2 becomes leader which has RemotePublicationEnabled = true, will do diff based on stale state instead of updated local state

Reply:
this needs more thinking when diff is applicable considering setting can be flipped in between as well.

Reply:
As of now, we haven't shipped the change regarding making remotePublicationEnabled a dynamic setting and edge cases regarding that can be discussed on #15478.
For now, we can keep 2 cases here:
1. Remote Publication is enabled: We ensure that while publication all following cluster manager nodes update the accepted state in their RemotePersistedState to the lastest state which is being published. In this case, every time a cluster manager commits the state quorum of the cluster manager would have accepted the state in RemotePersistedState and lagging cluster managers, will not be elected as their state version will be behind ours.
2. Now, if Remote Publication is disabled, we keep our old logic to upload the full state whenever a new cluster manager is trying to get elected.

Reply:
But RemotePublicationEnabled is a static setting what during rolling restart, the setting has different value across different nodes in the cluster

Reply:
> CM2 becomes leader which has RemotePublicationEnabled = true, will do diff based on stale state instead of updated local state

@shwetathareja Although lastManifest and lastState may be stale, the diff publication will not be incorrect. Because the diff will be published with lastState's stateUUID. Whichever node has the state which matches lastState's stateUUID will only be able to apply this diff while other node will download full state.

Reply:
So, we can't rely on cluster state settings to determine isPublicationEnabled here, as that will only be updated if setting change has been accepted by the quorum of nodes. How can we access the settings which are not updated in cluster state but have been modified on the node?

Reply:
> @shwetathareja Although lastManifest and lastState may be stale, the diff publication will not be incorrect. 

I would prefer to handle such cases upfront even before publication and assert localPersistedState and remotePersistedState are not diverged on active CM node otherwise it can cause unexpected race condition anytime.